src/sentry/api/endpoints/organization_details.py

```diff
@@ -954,7 +967,40 @@ def put(self, request: Request, organization) -> Response:
             with transaction.atomic(router.db_for_write(Organization)):
                 organization, changed_data = serializer.save()
 
-            if "targetSampleRate" in changed_data:
+            sampling_mode = organization.get_option("sentry:sampling_mode", SAMPLING_MODE_DEFAULT)
+            is_org_mode = sampling_mode == DynamicSamplingMode.ORGANIZATION.value
+            is_project_mode = sampling_mode == DynamicSamplingMode.PROJECT.value
+
+            if is_org_mode and "targetSampleRate" in changed_data:
+                organization.update_option(
+                    "sentry:target_sample_rate", serializer.validated_data["targetSampleRate"]
+                )
+                boost_low_volume_projects_of_org_with_query.delay(organization.id)
```

Comment:
To be extra safe, we should launch this task even if there's no `targetSampleRate` in changed data. We should do this every time the sampling mode changes and is now org mode.

Reply:
Yep, this still happens when the org mode changes anyways: https://github.com/getsentry/sentry/pull/80183/files#diff-a7787f64a5381d240074570d08806ce5559ff79652d9ecb9ba82a1e061395390R1004